src/combinatorics/quiver/basic.lean

```diff
@@ -86,17 +86,29 @@ def comp {U : Type*} [quiver U] {V : Type*} [quiver V] {W : Type*} [quiver W]
 { obj := λ X, G.obj (F.obj X),
   map := λ X Y f, G.map (F.map f), }
 
+@[simp] lemma comp_id {U : Type*} [quiver U] {V : Type*} [quiver V] (F : prefunctor U V) :
+  F.comp (id _) = F := by { cases F, refl, }
+
+@[simp] lemma id_comp {U : Type*} [quiver U] {V : Type*} [quiver V] (F : prefunctor U V) :
+  (id _).comp F = F := by { cases F, refl, }
```

Comment:
As these aren't definitional, I'm dubious these are a good idea. (They lead you into dependent type theory hell, and aren't particularly useful anyway?)

Reply:
Are you dubious about making them `simp` lemmas, or about even having those lemmas?

Reply:
@semorrison My understanding is that Remi is working on quivers for the purpose of having a typeclass-based graph structure (with `prefunctor` the morphism type for graphs) and the usual ways we work around dependent type hell in category theory don't really work in this context -- we don't have identity morphisms we can compose with for example, since quivers don't have composition. Equalities on morphisms become a necessary evil.

Here's the sort of hell we're in for: [simple_graph.walk.map_eq_of_eq](https://leanprover-community.github.io/mathlib_docs/combinatorics/simple_graph/connectivity.html#simple_graph.walk.map_eq_of_eq) (an application of this is if you have an isomorphism of graphs -- whose definition requires equalities on vertices! -- then you can define an equivalence on walk types).

If you don't want these in the main simp set, I'd be for them going into a new custom quiver simp set.

Reply:
No, let's try to them out! Walking into DTT hell with your eyes open is of course allowed. :-)